### PR TITLE
Added User Guide for device CRD

### LINKED ISSUE #722: User guide for device CRD
**What would you like to be added**:
There should be a user guide available for device CRD feature

**Why is this needed**:
A user guide would help new users understand the usage of the feature.

### PR DESCRIPTION
**Pull request by sujithsimon22**
**What type of PR is this?**
 /kind documentation

**What this PR does / why we need it**:
This PR contains a user guide for device CRD 

**Which issue(s) this PR fixes**:
Fixes #722

### CODE REVIEW

File: docs/guides/device_crd_guide.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+KubeEdge supports device management with the help of Kubernetes [CRDs](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/#customresourcedefinitions) and a Device Mapper (explained below) corresponding to the device being used.
+We currently manage devices from the cloud and synchronize the device updates between edge nodes and cloud, with the help of device controller and device twin modules.
+ 
+ 
+ ## Device Model
+
+A `device model` describes the device properties exposed by the device and property visitors to access these properties. A device model is like a reusable template using which many devices can be created and managed.
+
+Details on device model definition can be found [here](https://github.com/kubeedge/kubeedge/blob/master/docs/proposals/device-crd.md#device-instance-type-definition).
```
Comment (rohitsardesai83) [inline docs/guides/device_crd_guide.md]: https://github.com/kubeedge/kubeedge/blob/master/docs/proposals/device-crd.md#device-instance-type-definition  -->
https://github.com/kubeedge/kubeedge/blob/master/docs/proposals/device-crd.md#device-model-type-definition

File: docs/guides/device_crd_guide.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+A sample device model can be found [here](https://github.com/kubeedge/kubeedge/blob/master/docs/proposals/device-crd.md#device-instance-sample)
```
Comment (rohitsardesai83) [inline docs/guides/device_crd_guide.md]: https://github.com/kubeedge/kubeedge/blob/master/docs/proposals/device-crd.md#device-instance-sample  -->

https://github.com/kubeedge/kubeedge/blob/master/docs/proposals/device-crd.md#device-model-sample

File: docs/guides/device_crd_guide.md (hunk 1 above)
Comment (sujithsimon22) [reply docs/guides/device_crd_guide.md]: oops....my bad, thanks

File: docs/guides/device_crd_guide.md (hunk 2 above)
Comment (sujithsimon22) [reply docs/guides/device_crd_guide.md]: Done, thanks

**Pull request merged**